webcomponents/tool/src/main/frontend/js/sakai-search.js

```diff
@@ -90,51 +90,50 @@ class SakaiSearch extends SakaiElement {
   render() {
 
     return html`
-    <div class="input-group position-relative" role="search">
-    <input type="search"
+    <form class="input-group position-relative" onsubmit="return false">
+      <input type="search"
         class="sakaiSearch form-control"
         autocomplete="off"
         id="sakai-search-input"
+        pattern=".{3,}" title="${this.i18n.search_min_length}"
         placeholder="${this.i18n.search_placeholder}"
         @keydown=${this.search}
         @click=${this.handleSearchClick}
         value=${this.searchTerms}
-        aria-label="${this.i18n.search_placeholder}" />
-        <button class="btn btn-primary" type="button" id="sakai-search-button"  @click=${this.handleButtonClick}>Search Sakai</button>
-    </div>
+        aria-label="${this.i18n.search_placeholder}" 
+      />
+      <button class="btn btn-primary" type="submit" id="sakai-search-button"  @click=${this.handleButtonClick}>
+        Search Sakai
+      </button>
+    </form>
   ${this.noResults ? html`
-  <li class="list-group-item d-flex justify-content-between align-items-start">
-    <span class="no-results">No results found :(</span>
-  </li>  
+    <div class="list-group-item d-flex justify-content-between align-items-start">
+       <span class="no-results">No results found :(</span>
+    </div>
   ` : ""}
   ${this.results.length > 0 ? html`
     <button type="button" class="btn-close" aria-label="Close" @click=${this.closeResults}></button>
-      ${this.currentPageOfResults.map(r => html`
-      <li class="list-group-item d-flex justify-content-between align-items-start">
```

Comment:
If we could wrap the links in `<li>`'s that would be nice for better a11y. This patttern worked good, when I tested it.
```
<ul class="list-group">
    <li class="list-unstyled">
        <a href="#" class="list-group-item list-group-item-action">A search result</a>
    </li>
    <li class="list-unstyled">
        <a href="#" class="list-group-item list-group-item-action">A search result</a>
    </li>
    <li class="list-unstyled">
        <a href="#" class="list-group-item list-group-item-action">A search result</a>
    </li>
    <li class="list-unstyled">
        <a href="#" class="list-group-item list-group-item-action">A search result</a>
    </li>
    <li class="list-unstyled">
        <a href="#" class="list-group-item list-group-item-action">A search result</a>
    </li>                
</ul>
```

Reply:
I understand it, But for now, I wouldn't change it because it won't work with the keyboard(I tested). Therefore, I would suggest we go ahead for now, and after catching up on the deadline, we can come back and fix it.